### ISSUE TITLE
Spatial 'look_at' strange behaviour

### ISSUE THREAD

**Issue by Albatros53113** [godotengine/godot#37165]
**Godot version: 3.2.1**

**OS/device including version: Windows 10**

**Issue description: See example project**

**Steps to reproduce: See example project**

**Minimal reproduction project: https://github.com/Albatros53113/Turn-Zylinder/tree/master**

The issue is best described by looking at the example project 'readme'. Sorry if this is not an issue and I missunderstood some concepts, but seems like 'look_at' changes the axis it looks with.

**Comment by KoBeWi** [comment]
Is it maybe related to #35880?

**Comment by groud** [comment]
Sorry, but it's mandatory you try to explain what your problem is. Not everyone has the time to open, test you project, and try to understand what you felt was wrong.

**Comment by Albatros53113** [comment]
Yes I know. Tried really hard to cut it down to a simple, reproducable example, but wasn't able to do so. Found an functional error on my side, but that didn't help with the 'look_at' problem. As I print out a (local) 'look_at' vector and the 'basis.z' vector straight before the 'look_at' function and they are perfectly aligned, all I can say is that in my example project there never should be a turn on a mouseclick.
I have another solution that works now, so this is for someone who is interested in a possible problem. It's really simple code.

From this other solution I raised into another problem, that gives me some headache. Don't want to open another issue here, as I don't want to jangle s.o. nerves with minor problems that perhaps are only mine: The Vector3.dot() function sometimes gives values that are >1. Would be nice if this cases were captured, as this leads to poor errors that costs time.
Reproducable with:

func _ready():
	var vec1 = Vector3(0.061474, 0, 0.998109)
	var vec2 = Vector3(0.061474, 0, 0.998109)
	var scal = vec1.dot(vec2)
	print(str(scal))
	var angle = acos(scal)
	print(str(angle))
